File: frontend/src/shared/schemas/class.ts
import { z } from "zod";

// schema ห้องเรียน + ครูที่ปรึกษา + นักเรียนในห้อง — ตรงกับ backend (รายเทอม)

export const classListItemSchema = z.object({
  id: z.string(),
  grade_level: z.string(),
  room_name: z.string(),
  student_count: z.number(),
  advisor_count: z.number(),
});
export type ClassListItem = z.infer<typeof classListItemSchema>;
export const classListSchema = z.array(classListItemSchema);

export const classDetailSchema = z.object({
  id: z.string(),
  grade_level: z.string(),
  room_name: z.string(),
  semester_id: z.string(),
  created_at: z.string(),
});
export type ClassDetail = z.infer<typeof classDetailSchema>;

export type ClassBody = { grade_level: string; room_name: string };

export const classAdvisorSchema = z.object({
  id: z.string(),
  personnel_id: z.string(),
  prefix: z.string(),
  first_name: z.string(),
  last_name: z.string(),
});
export type ClassAdvisor = z.infer<typeof classAdvisorSchema>;
export const classAdvisorListSchema = z.array(classAdvisorSchema);

export type AddAdvisorBody = { personnel_id: string };

export const enrollmentSchema = z.object({
  id: z.string(),
  student_id: z.string(),
  student_no: z.number().nullable(),
  student_code: z.string(),
  prefix: z.string(),
  first_name: z.string(),
  last_name: z.string(),
});
export type Enrollment = z.infer<typeof enrollmentSchema>;
export const enrollmentListSchema = z.array(enrollmentSchema);

export type EnrollBody = { student_id: string; student_no?: number };

File: backend/internal/domain/class.go
package domain

import "time"

// Class — ห้องที่ปรึกษา (รายเทอม)
type Class struct {
	ID         string
	SchoolID   string
	SemesterID string
	GradeLevel string
	RoomName   string
	// นับจากการ join (สำหรับรายการ)
	StudentCount int
	AdvisorCount int
	CreatedAt    time.Time
	UpdatedAt    time.Time
}

type NewClass struct {
	GradeLevel string
	RoomName   string
}

type UpdateClass struct {
	GradeLevel string
	RoomName   string
}

// ClassAdvisor — ครูที่ปรึกษาของห้อง (M:N) พร้อมข้อมูลบุคลากรที่ join
type ClassAdvisor struct {
	ID          string
	PersonnelID string
	Prefix      string
	FirstName   string
	LastName    string
}

// ClassEnrollment — นักเรียนในห้อง (รายเทอม) พร้อมข้อมูลนักเรียนที่ join
type ClassEnrollment struct {
	ID          string
	StudentID   string
	StudentNo   *int
	StudentCode string
	Prefix      string
	FirstName   string
	LastName    string
}

// StudentClassBrief คือผลค้นหานักเรียน + ห้องที่สังกัด (ข้อมูลพื้นฐานเท่านั้น — PDPA)
type StudentClassBrief struct {
	StudentID   string
	StudentCode string
	Prefix      string
	FirstName   string
	LastName    string
	GradeLevel  string
	RoomName    string
}

type NewEnrollment struct {
	StudentID string
	StudentNo *int
}
